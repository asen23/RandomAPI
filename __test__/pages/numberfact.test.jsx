import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import NumberFact from "pages/numberfact/index"

const dummyNumberFact = ["a fact about number", "another fact about number"]

describe("NumberFact", () => {
    beforeEach(() => {
        fetch.resetMocks()
    })

    it("renders unchanged", async () => {
        fetch.mockResponseOnce(dummyNumberFact[0])

        const { asFragment } = render(<NumberFact />)
        expect(asFragment()).toMatchSnapshot()

        await screen.findByText(dummyNumberFact[0], { exact: false })

        expect(asFragment()).toMatchSnapshot()
    })

    it("gives new numberFact numberFact", async () => {
        fetch.mockResponses(...dummyNumberFact.map((e) => JSON.stringify(e)))

        render(<NumberFact />)

        const firstNumberFact = await screen.findByText(dummyNumberFact[0], {
            exact: false,
        })
        expect(firstNumberFact).toBeInTheDocument()

        const button = screen.getByRole("button")
        userEvent.click(button)

        const secondNumberFact = await screen.findByText(dummyNumberFact[1], {
            exact: false,
        })
        expect(secondNumberFact).toBeInTheDocument()
    })
})
