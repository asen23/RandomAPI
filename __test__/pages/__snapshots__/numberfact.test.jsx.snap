// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberFact renders unchanged 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center flex-col min-h-screen gap-8 bg-primary"
  >
    <h1
      class="text-4xl font-bold"
    >
      Random Number Fact
    </h1>
    <p>
      Loading...
    </p>
    <button
      class="border border-gray-400 px-2 py-1 rounded-lg hover:bg-white"
    >
      Get random number fact
    </button>
  </div>
</DocumentFragment>
`;

exports[`NumberFact renders unchanged 2`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center flex-col min-h-screen gap-8 bg-primary"
  >
    <h1
      class="text-4xl font-bold"
    >
      Random Number Fact
    </h1>
    <p>
      a fact about number
    </p>
    <button
      class="border border-gray-400 px-2 py-1 rounded-lg hover:bg-white"
    >
      Get random number fact
    </button>
  </div>
</DocumentFragment>
`;
